Add survey list delete dialog

marshmalien opened the pull request:
https://issues.redhat.com/browse/AAP-22463

![survey_delete](https://github.com/ansible/ansible-ui/assets/15881645/4bf898ef-0529-495d-8ba1-fac032aea416)

jerabekjiri commented:
Deleting from kebab menu in row actions shows TODO
![Screenshot from 2024-04-09 14-26-36](https://github.com/ansible/ansible-ui/assets/19647757/f59a1d45-18fe-43ab-a8b0-7c993c881ec6)

jerabekjiri reviewed (approved):
Other than the missing delete action in the kebab row actions, LGTM :+1: :rocket:

resoluteCoder reviewed (approved):
Pulled and tested locally echo what @nixocio said about kebab. lgtm :tada:

marshmalien commented on frontend/awx/resources/templates/hooks/useDeleteSurveyDialog.tsx:
```diff
@@ -0,0 +1,116 @@
+              compact
+              errorStateTitle="Error"
+              emptyStateTitle="No items"
+              autoHidePagination={true}
```
Yes, I'll add translations there.

marshmalien commented:
@jerabekjiri @resoluteCoder I added a second commit to address the row action kebab delete
@nixocio I updated the dialog styles and marked the title strings for translation 

<img width="1430" alt="Screenshot 2024-04-09 at 11 05 21 AM" src="https://github.com/ansible/ansible-ui/assets/15881645/7d7856d4-4190-409f-8f4e-d50c161688d3">


![row_delete](https://github.com/ansible/ansible-ui/assets/15881645/b7d60206-48ee-4841-af7b-b0b2c85135a9)

resoluteCoder commented:
Sweet works great! :+1:

nixocio reviewed (approved):
Good job @marshmalien
